front/tests/utils/index.ts

```diff
@@ -137,3 +43,15 @@ export async function extractNumberFromString(input: string): Promise<number> {
   const match = input.match(/\d+/);
   return match ? parseInt(match[0], 10) : 0;
 }
+// Utility function to read JSON files
+export const readJsonFile = (path: string) => JSON.parse(fs.readFileSync(path, 'utf8'));
+
+// Helper function to handle API error responses
+export async function handleApiResponse(
+  response: APIResponse,
+  errorMessage: string
+): Promise<void> {
+  if (!response.ok()) {
+    throw new Error(`${errorMessage}: ${response.status()} ${response.statusText()}`);
+  }
+}
```

Comment:
Some questions:
- why is this function async ? I think the request has already finished
- can't we return the response.json() ? This way we don't have to manipulate APIResponse everywhere

```suggestion
export function handleApiResponse(
  response: APIResponse,
  errorMessage: string
) {
  if (!response.ok()) {
    throw new Error(`${errorMessage}: ${response.status()} ${response.statusText()}`);
  }
  return response.json()
}
```

Reply:
This function handles only the error case of the API response to prevent code duplication. It's part of a larger `async `function, and adding `async` ensures consistency and allows for awaiting the response when the function is called

Reply:
- I removed the async function 
- The response.json() is used only once, so I didn't add it to the method (The response is handled directly within an if condition the second time).